google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/stub/metrics/Util.java

```diff
@@ -197,4 +214,46 @@ static void recordMetricsFromMetadata(
     // Record gfe metrics
     tracer.recordGfeMetadata(latency, throwable);
   }
+
+  // This function is to calculate the QPS based on current CPU
+  static double calculateQpsChange(double[] tsCpus, double target, double currentRate) {
```

Comment:
I'm not sure if this method fits here, maybe it can be a private method in RateLimitingStats